test/test_other.py

```diff
@@ -10265,6 +10265,40 @@ def test_wasm_features_section(self, args):
     self.run_process([EMCC, test_file('hello_world.c'), '-O2'] + args)
     self.verify_custom_sec_existence('a.out.wasm', 'target_features', False)
 
+  def test_wasm_features(self):
+    # Test that wasm features are explicitly enabled or disabled based on target engine version
+    def verify_features_sec(feature, expect_in):
+      with webassembly.Module('hello_world.o') as module:
+        features = module.get_target_features()
+      if expect_in:
+        self.assertTrue(feature in features and features[feature] == webassembly.TargetFeaturePrefix.USED)
+      else:
+        self.assertFalse(feature in features)
+
+    def compile(flags):
+      self.run_process([EMCC, test_file('hello_world.c'), '-c'] + flags)
+
+    compile([])
+    verify_features_sec('bulk-memory', False)
+    verify_features_sec('nontrapping-fptoint', False)
+    verify_features_sec('sign-ext', True)
+    verify_features_sec('mutable-globals', True)
+    verify_features_sec('multivalue', True)
+    verify_features_sec('reference-types', True)
+
+    compile(['-mnontrapping-fptoint'])
+    verify_features_sec('nontrapping-fptoint', True)
+
+    compile(['-sMIN_SAFARI_VERSION=150000'])
```

Comment:
Can you add some comments on what this is supposed to mean, like, Safari currently does or does not support such and such at version X? The same for line 10299.